test/server/chat-commands/moderation.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Tests for moderation commands
+ * @author Annika
+ */
+
+'use strict';
+
+const assert = require('assert').strict;
+const moderation = require('../../../.server-dist/chat-commands/moderation');
+
+const userUtils = require('../../users-utils');
+const User = userUtils.User;
+const Connection = userUtils.Connection;
+
+function makeUser(name, ip) {
+	const user = new User(new Connection(ip));
+	user.forceRename(name, true);
+	Users.users.set(user.id, user);
+	return user;
+}
+
+describe('room promotions', () => {
+	before(() => {
+		Rooms.global.addChatRoom('Promotion Testing');
+		this.room = Rooms.get('promotiontesting');
+
+		this.user = makeUser('Annika', '127.0.0.1');
+		this.user.setGroup('&');
+
+		this.targetUser = makeUser('Heart of Etheria', '127.0.0.1');
+	});
+
+	it('should not promote users with usernames >18 characters long', () => {
+		assert.throws(() => moderation.runPromote(this.user, this.room, 'a'.repeat(19), '+', undefined, true));
+		assert.doesNotThrow(() => moderation.runPromote(this.user, this.room, 'a'.repeat(18), '+', undefined, true));
+	});
```

Comment:
Err wait why?

Reply:
It seems like long usernames should either be allowed or banned, "allowed but can't be promoted" seems like an unnecessary condition to me.

Reply:
People can use `/forceroompromote` to promote usernames that are longer than 18 characters, and thus not allowed, but since we are able to forcibly promote users who are offline and unregistered, this check makes sense so you can't promote some username who cannot possibly log in.

Reply:
Oh, I see.